src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
+}
 
+function roundDown(num: number): number {
+  return Math.floor(num * 100) / 100;
 }
 
-export class Triangle {
+function throwError(...sides: number[]): void {
+  const firstTrue: boolean = sides.some((side: number) => side <= 0);
 
+  if (firstTrue) {
+    throw new Error('Invalid data for this figure!');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
```

Comment:
You can convert type Shape into Enum and avoid double writing.
in 23 it will like `shape = Shape.Triangle;`